### ISSUE TITLE
Document TS for common methods better

### ISSUE THREAD

**Issue by bahmutov** [cypress-io/cypress#1117]
Cypress 1.4.0

I looked at the IntelliSense popups in the project https://github.com/cypress-io/cypress-example-todomvc and the results are not very useful. While they could be useful (there is some formatting that I had to adjust on `cy.type` example to get this to work) like for `cy.type`

<img width="679" alt="cytype" src="https://user-images.githubusercontent.com/2212006/34263204-b322c288-e63c-11e7-971f-770349f70e34.png">

Many other code popups were useless.

For example, `cy` itself needs to be documented

<img width="509" alt="cyitself" src="https://user-images.githubusercontent.com/2212006/34263245-d1de58a4-e63c-11e7-800b-7efe521debf2.png">

`cy.visit` is ok, but the example needs to be tweaked to be displayed properly. Here is what works

```
/**
 * Type into a DOM element.
 *
 * @see https://on.cypress.io/type
 * @example
 *
 *    // type "Hello, World" into the <input> element
 *    cy.get('input').type('Hello, World')
 */
type...
```
The example is indented without any other text

<img width="687" alt="cyvisit" src="https://user-images.githubusercontent.com/2212006/34263289-ec29d6c0-e63c-11e7-9ce5-473ee89b2624.png">

`cy.get` is scary

<img width="697" alt="cyget" src="https://user-images.githubusercontent.com/2212006/34263360-36f0c92a-e63d-11e7-9350-a47b55d2e094.png">

`cy.should('have.class'...)` and other assertions need documentation

<img width="774" alt="cyshouldhaveclass" src="https://user-images.githubusercontent.com/2212006/34263389-5162a97c-e63d-11e7-89ce-8da02b9e3371.png">

**Comment by bahmutov** [comment]
I think if I can make `cypress-example-todomvc` have good helpful intellisense it will cover the basics.

**Comment by bahmutov** [comment]
much better

<img width="842" alt="screen shot 2017-12-22 at 11 22 11 am" src="https://user-images.githubusercontent.com/2212006/34305067-8697cc08-e70a-11e7-8337-a91dbc84ae7a.png">

**Comment by brian-mann** [comment]
Released in [`1.4.1`](https://docs.cypress.io/guides/references/changelog.html#1-4-1).
